storage/innobase/xtrabackup/test/inc/common.sh

```diff
@@ -885,6 +885,25 @@ function require_tokudb()
     fi
 }
 
+########################################################################
+# Return 0 if the xtrabackup has AddressSanitizer support
+########################################################################
+function is_asan()
+{
+    ldd $XB_BIN | grep -q libasan
+    return $?
+}
+
+#########################################################################
+# Skip test if xtrabackup has AddressSanitizer support
+########################################################################
+function skip_if_asan()
+{
+    if is_asan; then
```

Comment:
is_asn will return 0 if it has asan support right?
so this should be !is_asan

Reply:
@satya-bodapati is the other way around.

`grep -q libasan` will return 0 in case it finds a line from ldd corresponding to libasan, and will return 1 in case it doesn't find.


Here is one example of it:

```
main.xb_log_archiving                    w3	[skipped]   Incompatible with AddressSanitizer
```

Reply:
So it will be

if 0; then
  skip_test "Incompatible with Address Sanitizer"
fi

Reply:
correct.

Reply:
Hmm, when we detect, xtrabackup is built with ASAN, we don't want to skip? :thinking: this is not the intention right?

Reply:
When we detect it's built with asan support, it will be skipped. Here is one manual example:

No ASAN support:
```
marcelo@marce-bld:/work/pxb/ins/8.0/xtrabackup-test$ XB_BIN=/work/pxb/ins/8.0/bin/xtrabackup
marcelo@marce-bld:/work/pxb/ins/8.0/xtrabackup-test$ ########################################################################
marcelo@marce-bld:/work/pxb/ins/8.0/xtrabackup-test$ # Return 0 if the xtrabackup has AddressSanitizer support
marcelo@marce-bld:/work/pxb/ins/8.0/xtrabackup-test$ ########################################################################
marcelo@marce-bld:/work/pxb/ins/8.0/xtrabackup-test$ function is_asan()
> {
>     ldd $XB_BIN | grep -q libasan
>     return $?
> }
marcelo@marce-bld:/work/pxb/ins/8.0/xtrabackup-test$
marcelo@marce-bld:/work/pxb/ins/8.0/xtrabackup-test$ #########################################################################
marcelo@marce-bld:/work/pxb/ins/8.0/xtrabackup-test$ # Skip test if xtrabackup has AddressSanitizer support
marcelo@marce-bld:/work/pxb/ins/8.0/xtrabackup-test$ ########################################################################
marcelo@marce-bld:/work/pxb/ins/8.0/xtrabackup-test$ function skip_if_asan()
> {
>     if is_asan; then
> echo "Incompatible with AddressSanitizer"
> fi
> }
marcelo@marce-bld:/work/pxb/ins/8.0/xtrabackup-test$ ldd ${XB_BIN} | grep libasan
marcelo@marce-bld:/work/pxb/ins/8.0/xtrabackup-test$
marcelo@marce-bld:/work/pxb/ins/8.0/xtrabackup-test$ skip_if_asan
marcelo@marce-bld:/work/pxb/ins/8.0/xtrabackup-test$
```
Not skipping the test.

With ASAN support:

```
marcelo@marce-bld:/work/pxb/src/8.0$ ldd ${XB_BIN} | grep libasan
	libasan.so.5 => /lib/x86_64-linux-gnu/libasan.so.5 (0x00007f5c22cd1000)
marcelo@marce-bld:/work/pxb/src/8.0$ skip_if_asan
Incompatible with AddressSanitizer
marcelo@marce-bld:/work/pxb/src/8.0$
```

Reply:
ok, I didn't come out of this C based if statement. if 0 in C would enter the else branch. :headbang: